src/DeviceInput/InputDevices/deviceSourceManager.ts

```diff
@@ -109,6 +110,11 @@ export class DeviceSourceManager implements IDisposable {
      * @returns DeviceSource object
      */
     public getDeviceSource<T extends DeviceType>(deviceType: T, deviceSlot?: number): Nullable<DeviceSource<T>> {
+        if (this._checkForConnectedGamepads) {

```

Comment:
I definitely think this wouldbe more at home in the DeviceSourceManager constructor since:
1) you're using a private class member for `this._checkForConnectedGamepads`
2) it shouldn't really affect the logic of `getDeviceSource` device source manager, since we can now  have an additional precondition guaranteeing that any existing devices will be enumerated if we construct one with this flag as a parameter.
3) it avoids awkward situations where we may want to disable this without needing to fiddle with class members (just pass it into the constructor) and it prevents us from overloading getDeviceSource to also enumerate devices as well...

Reply:
Just an update, after talking with Gary and testing some things, I'll be putting this call in the constructor. The original idea for my choice was that I felt like custom callbacks should happen before devices are added to the DSM. I wanted the users to be able to handle controller context switching when a device is connected (eg. showing Xbox controls when an xbox contorller is connected) but looking at it now, I don't think that this would properly support that scenario. TBH, we should probably leave that to the users and keep things simple.

Reply:
tl;dr - You're right and I just needed to sleep on it.